### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by Zhabonchik**
Refactoring

- Restructure packages in main
- Create abstract parent class for tests
- Move controller building to abstract class in IT
- Expose 'loggers' endpoint in actuator

### CODE REVIEW

**Comment by SN4NTR** [comment]
- **mta_dev** folder -> **mta-extensions**
- mta extesnions must be called using the following pattern **mta-{env}.mtaext** - `mta-dev.mtaext`
- application should have 3 `application.yaml` files: for default profile - `application.yaml`, for local profile - `application-local.yaml` and for cloud profile - `application-cloud.yaml`
- `security-entity.user...` in **application.yaml** must be moved to **application-local.yaml**, as it's related to local env only
- `info.app...` in **application.yaml** should be on the top of this file in order to give information to other developers when they open file without scrolling down
- jpa related properties (e.g. database-platform) must be defined in dedicated profile file - `database-platform: org.hibernate.dialect.PostgreSQLDialect` is related to local env only but it's placed in `application.yaml` -> move to application-local.yaml`
- security package should be moved to `core` package
- `AuthToken` class that is used in `Destination` must be renamed to `DestinationAuthToken` and moved to `model` package in `destinationservice` package as it's not common auth token response. The same for `HttpHeader`
- `cloud` package should be either renamed to `btp` or create `btp` package in `cloud`
- Classes with private constructors can use lombok's `@NoArgsConstructor(access = PRIVATE)` instead of explicit constructor definition
- `DataFormatUtils` -> `DateFormatUtils`
- Requests for getting all entities must return `PagedModel` instead of `Page`
- `HttpConstantUtils` contains mix of different constants that are related to different features (cache, API, security and so on). It should be splitter based on feature
- It's redundant to comment each line of version in `<properties> in pom.xml`, just keep empty lines between them
- there is no need to define version for lombok in pom.xml, version is taken from parent
- you should have only 1 mustache file, that contains template for notification message. And you should create message bundle with translated values (e.g. messages_en.properties, messages_ru.properties) and use them during template creation in code (you can check it also with your colleagues)
- `docker-compose` file should be placed in `docker` folder in root of project
- I didn't get idea of `StudentId` class (and other similar). Looks like it's redundant
- postman collection is missed

File: pom.xml [hunk 1]
```diff
@@ -17,17 +17,25 @@
...
+        <!-- DB Migration -->
+        <liquibase-hanadb.version>4.31.1</liquibase-hanadb.version>
+        <!-- Annotation processing -->
         <lombok.version>1.18.38</lombok.version>
+        <!-- Template resolver -->
+        <mustache.version>0.9.14</mustache.version>
+        <!-- Mapping -->
         <mapstruct.version>1.6.3</mapstruct.version>
         <lombok-mapstruct.version>0.2.0</lombok-mapstruct.version>
+        <!-- Maven plugins -->
```
Comment (SN4NTR) [inline pom.xml]: these comments are redundant, jsut keep empty lines

File: src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import static com.leverx.learningmanagementsystem.utils.ITUtils.PAGE_SIZE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("dev")
+public abstract class AbstractIT {
```
Comment (SN4NTR) [inline src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java]: AbstractCommonIT

File: src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    protected MockMvc mockMvc;
+
+    @Autowired
+    protected ObjectMapper objectMapper;
+
+    @Autowired
+    protected StudentService studentService;
+
+    @Autowired
+    protected CourseService courseService;
```
Comment (SN4NTR) [inline src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java]: studentService and courseService are not commonly used across different tests, there is no need to put them here. Instead inject them in test classes where they are required

File: src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    public MockHttpServletRequestBuilder buildGetAllRequest(String path, String page, String pageSize) {
+        return get(path)
+                .param(PAGE, page)
+                .param(PAGE_SIZE, pageSize);
+    }
```
Comment (SN4NTR) [inline src/test/integration/com/leverx/learningmanagementsystem/AbstractIT.java]: move these methods with requests creation to separate class, e.g. `CourseRequestBuilder` that should be placed in `course.builder` package. The same for all other entities

File: src/test/integration/com/leverx/learningmanagementsystem/course/controller/CourseControllerIT.java [hunk 5]
```diff
@@ -25,50 +19,38 @@
...
-
-    @Autowired
-    SecurityEntityConfiguration configuration;
-
-    @Autowired
-    ObjectMapper objectMapper;
+class CourseControllerIT extends AbstractIT {
 
     @Test
+    @Tag("integration")
```
Comment (SN4NTR) [inline src/test/integration/com/leverx/learningmanagementsystem/course/controller/CourseControllerIT.java]: `@Tag("integration")` must be on class level instead of methods. In your case, just put in on AbstractCommonIT class

File: src/test/integration/com/leverx/learningmanagementsystem/course/controller/CourseControllerIT.java [hunk 6]
```diff
@@ -25,50 +19,38 @@
...
 
         // then
         response.andExpect(status().isOk())
                 .andExpect(jsonPath("$.content.size()").value(TOTAL_NUMBER_OF_COURSES));
     }
 
     @Test
+    @Tag("integration")
     @Sql(scripts = {CLEAN_SQL, INSERT_SQL}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
     @WithMockUser(roles = "USER")
```
Comment (SN4NTR) [inline src/test/integration/com/leverx/learningmanagementsystem/course/controller/CourseControllerIT.java]: `@WithMockUser(roles = "USER")` also can be put on class level. In case you need to test unauthorized access, put `@WithAnonymousUser` on this test method

File: src/test/unit/com/leverx/learningmanagementsystem/AbstractTest.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package com.leverx.learningmanagementsystem;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public abstract class AbstractTest {
```
Comment (SN4NTR) [inline src/test/unit/com/leverx/learningmanagementsystem/AbstractTest.java]: for unit tests there is no need to create common abstract class, it's only for integration tests to avoid recreation of context for each test

File: src/main/java/com/leverx/learningmanagementsystem/core/scheduling/SendCourseNotificationJob.java [hunk 8]
```diff
@@ -1,9 +1,9 @@
-package com.leverx.learningmanagementsystem.utils.scheduling;
+package com.leverx.learningmanagementsystem.core.scheduling;
```
Comment (SN4NTR) [inline src/main/java/com/leverx/learningmanagementsystem/core/scheduling/SendCourseNotificationJob.java]: it must be in `course.job` package, as it's related to courses feature

File: src/main/java/com/leverx/learningmanagementsystem/email/service/MailSenderProvider.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package com.leverx.learningmanagementsystem.email.mailsenderprovider;
+package com.leverx.learningmanagementsystem.email.service;
 
 import org.springframework.mail.javamail.JavaMailSender;
```
Comment (SN4NTR) [inline src/main/java/com/leverx/learningmanagementsystem/email/service/MailSenderProvider.java:4]: add empty lines between class name and methods. Check for all other classes

File: src/main/java/com/leverx/learningmanagementsystem/email/model/MailConfigUserProvided.java [hunk 10]
```diff
@@ -10,7 +10,7 @@
 @ConfigurationProperties("user-provided.mail.smtp")
 @Data
 @Profile("cloud")
-public class UserProvidedMailConfig implements MailConfig {
+public class MailConfigUserProvided implements MailConfig {
     @JsonProperty("mail.transport.protocol")
     private String protocol;
     @JsonProperty("mail.smtp.port")
```
Comment (SN4NTR) [inline src/main/java/com/leverx/learningmanagementsystem/email/model/MailConfigUserProvided.java]: empty lines between variables

File: pom.xml [hunk 11]
```diff
@@ -17,17 +17,25 @@
     <description>LearningManagementSystem</description>
 
     <properties>
+        <!-- Java -->
         <java.version>21</java.version>
+        <!-- Spring starters -->
+        <xsuaa-starter.verion>3.6.0</xsuaa-starter.verion>
```
Comment (vicras) [inline pom.xml]: typo: version

Comment (vicras) [reply pom.xml]: Postman still missed

Comment (Zhabonchik) [reply pom.xml]: It is in LearningManagementSystem.http file. Should it be imported in another way?

Comment (vicras) [reply pom.xml]: usually we use json for storing Postman collection 
image

Comment (vicras) [reply pom.xml]: and in a separate postman folder

**Comment by SN4NTR** [comment]
- `DestinationServiceUtils` -> `DestinationServiceConstants` (package also should be renamed to `constant`. The same for other similar classes, where only constants are stored
- cases like `!isNull(language)` can be rewritten with `nonNull(language)`. Check in all places
- constants in code must be `static final`, in some classes they are only `final` (e.g. in SendCourseNotificationJob)
- language folder in resources should be renamed to `i18n`
- PagedResourcesAssembler in controllers should be removed. PagedModel can be created in this way:
```
new PagedModel<>(dtos);
```
- lesson.dto contains pcakges like `ClassroomLesson` that violates java conventions. Capital letters are not allowed in package names. Either put all types of lessons into `dto` package or you can follow package by feature approach inside of lesson package:
image
- create `application-local.yaml` file and replace `dev` profile in code with `local`
- put `@Tag("unit")` on class level instead of each method
- classes in `utils` package should be splitter by features in unit and integration tests packages. E.g. move `CourseTestUtils` to package `course.common.utils` and similar for others
- in case you need some confidential credentials for local env and for testing, just keep this values empty before committing (check your mail configs)

File: src/test/unit/com/leverx/learningmanagementsystem/coursesettings/service/CourseSettingsServiceTest.java [hunk 12]
```diff
@@ -8,17 +8,19 @@
...
-import static com.leverx.learningmanagementsystem.utils.CourseSettingsTestUtils.EXISTING_COURSE_SETTINGS_ID;
-import static com.leverx.learningmanagementsystem.utils.CourseSettingsTestUtils.TOTAL_NUMBER_OF_COURSE_SETTINGS;
+import static com.leverx.learningmanagementsystem.coursesettings.common.utils.CourseSettingsTestUtils.EXISTING_COURSE_SETTINGS_ID;
+import static com.leverx.learningmanagementsystem.coursesettings.common.utils.CourseSettingsTestUtils.TOTAL_NUMBER_OF_COURSE_SETTINGS;
 import static org.junit.jupiter.api.Assertions.assertAll;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.when;
 
+@ActiveProfiles("test")
```
Comment (SN4NTR) [inline src/test/unit/com/leverx/learningmanagementsystem/coursesettings/service/CourseSettingsServiceTest.java]: `@ActiveProfiles("test")` is used in integration tests for building spring context with this profile. There is no need to use it in unit tests
